Add a write_log output plugin which logs values.

pyr opened the pull request:
I find this to be useful when developping new input plugins,
it allows creating a very simple configuration and combined
with `-f` can be used for a very simple workflow.

I'm proposing this, since I think it could be useful for
people wanting to debug their config on local agents as well.

katzj commented:
:+1: I tend to use the csv plugin for this but this looks nicer

katzj commented on src/write_log.c:
```diff
@@ -0,0 +1,76 @@
+    if (status != 0) /* error message has been printed already. */
+        return (status);
+
+    INFO ("write_log values:\n%s", buffer);
```
Info seems aggressive for this

pyr replied:
@katzj I'm open to ideas, but as soon as you start using this, you want to see the logs. DEBUG means compiling at `-DDEBUG` which for people just wanting to check that things are operating normally using standard packages it would become unfeasible.

katzj replied:
Hm, yeah.  I forgot that we compile our packages with -DDEBUG just because it makes it easier for me to turn it on for people.

mfournier commented on src/write_log.c:
```diff
@@ -0,0 +1,76 @@
+ * Copyright (C) 2015       Pierre-Yves Ritschard
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
```
Maybe switch to MIT ?

mfournier commented on src/collectd.conf.pod:
```diff
@@ -7129,6 +7129,12 @@ want to set B<metadata.broker.list> to your Kafka broker list.
 
 =back
 
+=head2 Plugin C<write_log>
```
Plugins without any configure options aren't usually mentioned in the manpage (which is something that could be discussed). On the other hand, they are listed in the README, which isn't the case yet for this plugin.

mfournier commented:
Seems useful, indeed. Just wondering why a new plugin rather than adding an option to the existing logfile, syslog, and log_logstash plugins ?

pyr commented:
@mfournier I switched to MIT and added a mention in the README. As for the separate plugin, it's just because it is not implementing the log callbacks and is independent of the log module that you will be using.
